Compare Three Prime and Exon Array Intensities and Ratios
==========================================================

```{r echo=FALSE}
util.dir <- file.path(Sys.getenv("AA"),"utils")
data.dir <- file.path(Sys.getenv("AA"),"data") 
source(paste(util.dir,"utilitiesSigTest.R",sep="/"))
```

*Data Directories* Different versions of normalized data sets 

Curated dataset, includes VERA SAM statistical testing
```{r}
dir3prime.cur <- paste(data.dir,"20120926.curated.3prime",sep="/")
```
All 3 prime arrays
```{r}
dir3prime.all <- paste(data.dir,"20121001.3prime",sep="/")
```

Exon LPS timecourse
```{r}
direxon.cur <- paste(data.dir,"20121001.curated.exon",sep="/")
```

All exon arrays
```{r}
direxon.all <- paste(data.dir,"20120928.exon",sep="/")
```

chipseq uses: 3prime.cur and exon.cur  
tfinf uses: 3prime.cur  
allarrays uses: 3prime.all, exon.all  

*Data Loading*

Load VERA SAM processed

```{r}
selectiveLoad("lps.lambdas",paste(dir3prime.cur,"all.lambdas.objects.RData",sep="/"))
selectiveLoad("lps.mus",paste(dir3prime.cur,"all.mus.objects.RData",sep="/"))
selectiveLoad("lps.ratios",paste(dir3prime.cur,"all.ratios.objects.RData",sep="/"))
```

Load non-stat tested
```{r}
load(paste(dir3prime.all,"dm.RData",sep="/"))
dm.3prime <- dm
load(paste(dir3prime.all,"CSSs.tc.RData",sep="/"))
CSSs.tc.3prime <- CSSs.tc

load(paste(direxon.all,"dm.RData",sep="/"))
dm.exon <- dm
load(paste(direxon.all,"CSSs.tc.RData",sep="/"))
CSSs.tc.exon <- CSSs.tc

load(paste(direxon.cur,"dm.RData",sep="/"))
dm.exon <- dm
load(paste(direxon.cur,"CSSs.tc.RData",sep="/"))
CSSs.tc.exon <- CSSs.tc
```
Note: order dependence above. We will use the curated Exon set below, containing LPS timecourse.

Probeset IDs on both platform
```{r}
onboth.ps <- intersect(rownames(dm.exon),rownames(lps.mus))
```

Ratios on exon arrays
```{r}
rats.exon <- log10(dm.exon[,2:ncol(dm.exon)]/dm.exon[,1])
```

Absolute values
---------------

```{r}
plot(lps.mus[onboth.ps,"min0"],dm.exon[onboth.ps,"BMDM_Bl6_LPS_0000___Female"],xlab="3 prime",ylab="exon",main="Intensity, T=0",log='xy')
```

Exon abs values 2-3 times higher than three prime.  Find more quantitative estimate esimate. Look at exon deciles over three-prime deciles. 
```{r}
quantile(probs=seq(0.5,0.95,0.05),x=dm.exon[onboth.ps,"BMDM_Bl6_LPS_0000___Female"])
quantile(probs=seq(0.5,0.95,0.05),x=lps.mus[onboth.ps,"min0"])
```
Would seem we would be most interested in quantiles above 70.


```{r}
dec.comp <- quantile(probs=seq(0.5,0.95,0.05),x=dm.exon[onboth.ps,"BMDM_Bl6_LPS_0000___Female"])/quantile(probs=seq(0.5,0.95,0.05),x=lps.mus[onboth.ps,"min0"])
dec.comp
quantile(probs=seq(0.5,0.95,0.05),x=dm.exon[onboth.ps,"BMDM_Bl6_LPS_0060___Female"])/quantile(probs=seq(0.5,0.95,0.05),x=lps.mus[onboth.ps,"min60"])
quantile(probs=seq(0.5,0.95,0.05),x=dm.exon[onboth.ps,"BMDM_Bl6_LPS_0120___Female"])/quantile(probs=seq(0.5,0.95,0.05),x=lps.mus[onboth.ps,"min120"])
quantile(probs=seq(0.5,0.95,0.05),x=dm.exon[onboth.ps,"BMDM_Bl6_LPS_0240___Female"])/quantile(probs=seq(0.5,0.95,0.05),x=lps.mus[onboth.ps,"hr4"])
```

Ratio of medians of each distribution
```{r}
median.ratio <- dec.comp["50%"]
median.ratio
```

Plot redone
```{r echo=FALSE}
plot(lps.mus[onboth.ps,"min0"],dm.exon[onboth.ps,"BMDM_Bl6_LPS_0000___Female"],xlab="3 prime",ylab="exon",main="Intensity, T=0",log='xy')
abline(0,1,col='blue')
abline(log10(median.ratio),1,col='red') # 50% quantile
abline(log10(dec.comp["90%"]),1,col='orange') # 90% quantile
abline(v=300,col='green')
abline(h=median.ratio*300,col='green')
abline(v=200,col='purple')
abline(h=median.ratio*200,col='purple')
```
Blue diagonal: slope of unity  
Red diagonal: ratio of medians  
Orange diagonal: ratio of 90% quantiles  
Green vertical: 300 units   
Green horizontal: 300 units, median converted ```r 300*median.ratio```  
Purple vertical: 200 units   
Purple horizontal: 200 units, median converted ```r 200*median.ratio```  



Can also try to look at distributions with kernel smoothing
```{r chache=T}
library(KernSmooth)
x <- cbind(log10(lps.mus[onboth.ps,"min0"]),log10(dm.exon[onboth.ps,"BMDM_Bl6_LPS_0000___Female"]))
est <- bkde2D( x, bandwidth=c(0.2,0.2) ) # larger bandwithds <-> wider kernel
contour(est$x1,est$x2,est$fhat,xlab="3 prime",ylab="exon",main="Log10(Intensity), T=0")
abline(0,1)
abline(log10(median.ratio),1)
```

Two peak structure is somewhat baffling. Slope close to kernel overall "axis"
Perhaps upper peak is the "true" expressed ones we care about ?


Correspondence of lambda.cutoff to ratio cutoff
-----------------------------------------------

Lambda cutoff in terms of ratios.  What ratio does the lamba.cutoff correspond to?  Let's look at the ratio of 2hrs to unstim.

```{r}
plot(lps.ratios[onboth.ps,"min120"],lps.lambdas[onboth.ps,"min120"])
lambda.cutoff <- 26.44526 ## 0.05% cutoff September 2012
abline(h=lambda.cutoff)
```

Ratio of nearest probeset to cutoff
```{r}
log10ratcut <- abs(lps.ratios[which.min(abs(lps.lambdas[,"min120"]-lambda.cutoff)),"min120"])
ratcut <- 10^log10ratcut
log10ratcut
ratcut
```
The value ```r ratcut``` is the ratio corresponding to lambda.cutoff (with numerator larger than denominator).

In principle, if we didn't have the statistics, we would use this ratio cutoff and get similar results.


Ratio comparison between platforms
----------------------------------

```{r}
par(mfrow=c(2,2))
plot(lps.ratios[onboth.ps,"min60"],rats.exon[onboth.ps,"BMDM_Bl6_LPS_0060___Female"],xlab="3 prime",ylab="exon",main="Log10 Ratios, T=1 hr")
abline(0,1)
plot(lps.ratios[onboth.ps,"min120"],rats.exon[onboth.ps,"BMDM_Bl6_LPS_0120___Female"], xlab="3 prime",ylab="exon",main="Log10 Ratios, T=2 hr")
abline(0,1)
plot(lps.ratios[onboth.ps,"hr4"],rats.exon[onboth.ps,"BMDM_Bl6_LPS_0240___Female"],xlab="3 prime",ylab="exon",main="Log10 Ratios, T=4 hr")
abline(0,1)
plot(lps.ratios[onboth.ps,"hr6"],rats.exon[onboth.ps,"BMDM_Bl6_LPS_0360___Female"],xlab="3 prime",ylab="exon",main="Log10 Ratios, T=6 hr")
abline(0,1)
```


When doing contour plots for T=2 hrs (not shown) the distributions are mostly circular!
Probably has to do with how few induced genes there are relative to whole

```{r eval=FALSE}
# analysis to explore later. Seems like it is hard to see more than cricular contours 
x <- cbind(lps.ratios[onboth.ps,"min120"],rats.exon[onboth.ps,"BMDM_Bl6_LPS_0120___Female"])
est <- bkde2D( x, bandwidth=c(0.2,0.2) ) # larger bandwithds <-> wider kernel
est <- bkde2D( x, gridsize = c(1000L, 1000L),bandwidth=c(0.2,0.2) ) # larger bandwithds <-> 
wider kernel
contour(est$x1,est$x2,est$fhat,xlab="3 prime",ylab="exon",main="Log10 Ratios, T=240",nlevels=50)
#Weird, elongation only happens when gridsize upped from default of 51L to 500L or more 
#And what is this L thing?
```

*Decile Plots*

```{r}
v3 <- quantile(x=lps.ratios[onboth.ps,"hr4"],probs=seq(0.1,0.9,0.1))
ve <- quantile(x=rats.exon[onboth.ps,"BMDM_Bl6_LPS_0240___Female"],probs=seq(0.1,0.9,0.1))
plot(v3,ve)
abline(0,1)
```
The highest percentile in this plot is 90. Fairly consistent in this range.


At the very highest percentiles, we do start to see some differences
```{r}
v3 <- quantile(x=lps.ratios[onboth.ps,"hr4"],probs=seq(0.01,0.99,0.01))
ve <- quantile(x=rats.exon[onboth.ps,"BMDM_Bl6_LPS_0240___Female"],probs=seq(0.01,0.99,0.01))
plot(v3,ve)
abline(0,1)
```

Here's the full QQ-plot, for hr 4
```{r}
qqplot(lps.ratios[onboth.ps,"hr4"],rats.exon[onboth.ps,"BMDM_Bl6_LPS_0240___Female"])
abline(0,1)
```
Thus the three prime ratio distribution has somewhat longer tails than the exon distribution.

Comparison of constitutively expressed probesets
-----------------------------------------------

Let's see if this changes the correspondence between constitutive counts on the two platforms

```{r echo=FALSE}
sigSlice <- function( lambdaCutoff , ratioMatrix, lambdaMatrix){
  whichOnes <- unique(row.names(which(lambdaMatrix>lambdaCutoff,arr.ind=TRUE)))
  ratioMatrix[whichOnes,]  
}
ncbiID <- as.character(unlist(sapply(rownames(lps.mus),strsplit,split="_at")))
names(ncbiID) <- rownames(lps.mus)
```



*Comparison of differentially expressed genes*

Three prime
```{r}
mu.cutoff <- 300
imax <- 8 ## imax=8 <-> 6 hrs 
lps.6hr.ps <- rownames(sigSlice(lambda.cutoff,lps.ratios[,1:(imax-1)],lps.lambdas[,1:(imax-1)]))
low.expressors <- names(which(apply(lps.mus[lps.6hr.ps,1:imax]<mu.cutoff,1,sum)==imax))
lps.6hr.ps <- setdiff(lps.6hr.ps,low.expressors)
```
Length of lps.6hr.ps is ```r length(lps.6hr.ps)```.

Exon    
Cutoff choice - absolute. Scale the thee prime cutoff using the top decile normalization
```{r}
abs.cutoff <- dec.comp["90%"] * mu.cutoff
```

We could use the one that was "mapped from" lambda.cutoff above, but we don't have triplicated measurements.   May have to stick with the 
```{r}
rat.cutoff <- 1.492107 
rat.cutoff <- 2

dm.lps.exon <- dm.exon
## nt=5 corresponds to 6 hrs
nt <- 5
maxabs <- apply(dm.lps.exon[,1:nt],1,max)
abs.logvec <- ( maxabs > abs.cutoff )
tcs <- dm.lps.exon
ratmat <- (tcs/tcs[,1])[,2:nt]
maxrats <- apply(ratmat,1,max)
minrats <- apply(ratmat,1,min)
rat.logvec <- ( maxrats > rat.cutoff ) | ( minrats < (1/rat.cutoff) )
diffexp.exon.ps <- names(which(abs.logvec & rat.logvec))
```
Length of diffexp.exon.ps is ```r length(diffexp.exon.ps)```.


Compare 
```{r}
compareSets(intersect(lps.6hr.ps,onboth.ps),intersect(diffexp.exon.ps,onboth.ps))
````

*Comparison of constitutive genes*

Three Prime
```{r}
mu.high.cutoff <- 300
high.expressors <- names(which(apply(lps.mus[,1:imax]>mu.high.cutoff,1,sum)==imax))
constitutive.3prime.ps <- setdiff(high.expressors,lps.6hr.ps)
```

Exon
```{r}
mu.high.cutoff <- abs.cutoff
high.expressors <- names(which(apply(dm.lps.exon[,1:nt]>mu.high.cutoff,1,sum)==nt))
constitutive.exon.ps <- setdiff(high.expressors,diffexp.exon.ps)
```


```{r}
compareSets(intersect(constitutive.3prime.ps,onboth.ps),intersect(constitutive.exon.ps,onboth.ps))
```
